### PR TITLE
[CPU] Improve instruction generating algorithm for pow(a, large_N)

### PR DESCRIPTION
**Pull request by xuchen-intel**
### Details:
 - *Improve instruction generating algorithm for pow(a, N), where N is very large. Besides, fix overflow during converting from float to int32.*

### Tickets:
 - *[154601](https://jira.devtools.intel.com/browse/CVS-154601)*

### CODE REVIEW

**Comment by xuchen-intel** [comment]
@dmitry-gorokhov Hi Dmitry! Could you please take a look?

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/convert_to_plugin_specific_node.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@ namespace test {
 
 using ConvertToPluginSpecificNodeParams = std::tuple<ov::Shape,                      // non const input shape
                                                      ov::Shape,                      // const input shape
+                                                     bool,                           // large const number
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/convert_to_plugin_specific_node.cpp:17]: Do we really need separate param? Can we extend the range for random generator by default?

File: src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp [hunk 2]
```diff
@@ -1576,10 +1576,27 @@ void jit_power_static_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs,
             }
         }
     } else if (std::floor(power) == power && power != 0) {
-        int ipower = std::abs(static_cast<int>(power));
+        int64_t ipower = std::abs(static_cast<int64_t>(power));
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp:1579]: Shouldn't we update aarch64 emitter as well?

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/convert_to_plugin_specific_node.cpp (hunk 1 above)
Comment (xuchen-intel) [reply src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/convert_to_plugin_specific_node.cpp]: Good point! Applied. Thanks Dmitry!

File: src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp (hunk 2 above)
Comment (xuchen-intel) [reply src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp]: Thanks Dmitry for the comment! I've checked that the aarch64 emitter is already correct. https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp#L1659-L1671
BTW, I've also aligned x64 algorithm with aarch64, which is more concise.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**